model_compression_toolkit/core/common/mixed_precision/kpi_tools/kpi_methods.py

```diff
@@ -133,6 +159,156 @@ def total_weights_activation_kpi(mp_cfg: List[int], graph: Graph, fw_info: Frame
     return np.array(weights_activation_memory)
 
 
+def bops_kpi(mp_cfg: List[int],
+             graph: Graph,
+             fw_info: FrameworkInfo,
+             fw_impl: FrameworkImplementation,
+             set_constraints: bool = True) -> np.ndarray:
+    """
+    Computes a KPIs vector with the respective bit-operations (BOPS) count for each configurable node,
+    according to the given mixed-precision configuration of a virtual graph with composed nodes.
+
+    Args:
+        mp_cfg: A mixed-precision configuration (list of candidates index for each configurable node)
+        graph: Graph object.
+        fw_info: FrameworkInfo object about the specific framework (e.g., attributes of different layers' weights to quantize).
+        fw_impl: FrameworkImplementation object with specific framework methods implementation.
+        set_constraints: A flag for utilizing the method for KPI computation of a
+            given config not for LP formalization purposes.
+
+    Returns: A vector of node's BOPS count.
+    Note that the vector is not necessarily of the same length as the given config.
+
+    """
+
+    if not set_constraints:
+        return _bops_kpi(mp_cfg,
+                         graph,
+                         fw_info,
+                         fw_impl)
+
+    virtual_bops_nodes = [n for n in graph.get_topo_sorted_nodes() if isinstance(n, VirtualActivationWeightsNode)]
+
+    mp_nodes = graph.get_configurable_sorted_nodes_names()
+    bops = [n.get_bops_count(fw_impl, fw_info, candidate_idx=_get_node_cfg_idx(n, mp_cfg, mp_nodes)) for n in virtual_bops_nodes]
```

Comment:
Shouldn't we compute BOPs for the interest points and not configurable nodes?

Reply:
No, interest points are only used for distance metric computation.
We need to compute the KPI for each node that contributes to it.